src/unix/luv/lwt_luv.ml

```diff
@@ -0,0 +1,72 @@
+let l f = function
+| Result.Error e -> failwith (Luv.Error.err_name e)
+| Ok l -> List.iter (function
+| `DISCONNECT -> ()
+| `PRIORITIZED -> ()
+| `READABLE -> f ()
+| `WRITABLE -> f ()
```

Comment:
This looks like it will call `f` twice if both `` `READABLE`` and `` `WRITABLE`` are returned. Is that intentional? Have you tried calling `f` only once, if either or both of those values are present? Is returning both even possible with the way the code is written? I suspect not, but I also suspect you have much more direct and recent experience with this now than I do :)

Also, if we always ignore `` `DISCONNECT`` and `` `PRIORITIZED`` (at least for now), why register for them later on down in this code?

Reply:
This is  a  leftover from  when I was debugging. I'll change it to just listen for the event it needs (`READABLE and `WRITABLE depending on function) and then just call the callback.

Reply:
And we should never get both `` `READABLE `` and `` `WRITABLE `` at the same  time since  we're explicitly passing in the mask we want.